packages/sidecar/src/mcp/index.ts

```diff
@@ -12,28 +11,112 @@ const NO_ERRORS_CONTENT: CallToolResult = {
   content: [
     {
       type: "text",
-      text: "No recent errors found in Spotlight. This might be because the application started successfully, but runtime issues only appear when you interact with specific pages or features.\n\nAsk the user to navigate to the page where they're experiencing the issue to reproduce it, that way we can get that in the Spotlight debugger. So if you want to check for errors again, just ask me to do that.",
+      text: `**No errors detected in Spotlight** (last 60 seconds)
+
+**This means:**
+- Application is currently running without runtime failures
+- No crashes, exceptions, or critical issues in the recent timeframe
+- System appears stable at the moment
+
+**Next debugging steps:**
+
+1. **If user reports a specific issue:**
+   - Ask them to reproduce the problem (click the button, submit the form, navigate to the page)
+   - Run this tool again immediately after they reproduce it
+   - Errors will appear in real-time as they happen
+
+2. **If investigating existing code:**
+   - Check application logs separately
+   - Look for TODO comments, error handling gaps, or potential edge cases in the code
+   - Consider testing error scenarios (invalid inputs, network failures, etc.)
+
+3. **Proactive error detection:**
+   - Have user interact with recently changed features
+   - Test API endpoints or database operations that might be fragile
+   - Check pages/features mentioned in recent commits
+
+** Pro tip:** Absence of errors doesn't mean absence of bugs - it just means no runtime failures occurred recently. The issue might be logical errors, UI problems, or dormant bugs waiting for specific conditions.`,
     },
   ],
 };
 
-export function createMcpInstance(buffer: MessageBuffer<Payload>) {
+export function createMcpInstance() {
   const mcp = new McpServer({
     name: "spotlight-mcp",
     version: String(process.env.npm_package_version),
   });
 
-  const errorsBuffer = new MessageBuffer<Payload>(10);
-  buffer.subscribe((item: Payload) => {
-    errorsBuffer.put(item);
-  });
-
-  mcp.tool(
-    "get_errors",
-    "Fetches the most recent errors from Spotlight debugger. Returns error details, stack traces, and request details for immediate debugging context.",
-    async () => {
-      const envelopes = errorsBuffer.read();
-      errorsBuffer.clear();
+  mcp.registerTool(
+    "get_local_errors",
+    {
+      title: "Get Local App Errors",
+      description: `Retrieve recent application errors from Spotlight debugger to diagnose failures, crashes, and exceptions across your entire stack - frontend, backend, and edge functions.
+
+**CRITICAL: Call this tool IMMEDIATELY when:**
+- User says "error", "broken", "not working", "failing", "crash", "bug", "issue"
+- You encounter errors during testing or code execution
+- Before and after making code changes to verify no regressions
+- Investigating any unexpected behavior or performance problems
+
+**What you get:**
```

Comment:
It would be ideal if we kept these changes to a separate PR. Won't block for this but thoughts for future PRs :)